Add board name customization and LED status

L1cardo opened the pull request:
1. Add board name customization
<img height="200" alt="image" src="https://github.com/user-attachments/assets/fdd9527b-730d-42f6-868e-a1ea9510ebc7" />

2. LED status to show the connectivity
	- Connected: Solid on
	- Disconnected: Blink for 60s and if still no connection then off

L1cardo committed:
add board name customization

L1cardo committed:
add led status

L1cardo commented:
Hi, can you merge this pr ? @pfeerick

pfeerick commented on src/main.cpp:
```diff
@@ -1,6 +1,23 @@
 #include <Adafruit_TinyUSB.h> 
 
+// Board name
+#define MANUFACTURER_NAME "USB_ELRS_Receiver"
+#define PRODUCT_NAME     "USB_ELRS_Receiver"
```
Why the underscores... are they not human readable labels, thus spaces permitted? 

I wonder if these would also be better in the platformio.ini as board level defines, as then the "product" name could be tied to the board used (i.e. ELRS Pico USB Receiver, ELRS RP2040 USB Receiver, etc ... or is that getting to convoluted?  Your thoughts? edit: and then it dawns on me that you just used the repo name, lol. 

What do you think about something more like "ELRS Community" for the Manufacturer name? It shouldn't really be the same as the Product name as it serves a different purpose.

pfeerick commented on src/main.cpp:
```diff
@@ -105,9 +119,15 @@ void debug_out();
+  // Custom board name
+  USBDevice.setManufacturerDescriptor(MANUFACTURER_NAME);
+  USBDevice.setProductDescriptor(PRODUCT_NAME);
+  USBDevice.setID(0x1209, 0x0408); // 0x1209 is common ID for opensource projects, 0x0408 is generated randomly
```
I'll put a request in to https://pid.codes/ for a proper PID assignment, rather than just  picking a random one since this is their VID. No need for you to comment on this, it's a TODO for me.

pfeerick commented on src/main.cpp:
```diff
@@ -1,6 +1,23 @@
+bool ledState = LOW;             // Current LED state
+
+#define LED_BLINK_INTERVAL 500    // LED blink interval when disconnected (ms)
+#define CONNECTION_TIMEOUT 60000  // Time to wait before turning off LED completely (ms)
```
I assume this 60 seconds because the default delay before going into wifi mode is 16 seconds?

pfeerick committed:
refactor(led): improve updateLed initialisation and reduce digitalWrite calls

pfeerick committed:
refactor: move board-specific config to platformio.ini

Replace #ifdef ARDUINO_RASPBERRY_PI_PICO blocks in main.cpp with
build flags (CRSF_SERIAL, LED_PIN, LED_ON, LED_OFF) defined per board
in platformio.ini, supporting active-low LEDs on XIAO SAMD21/RP2040
and active-high on Pico.

pfeerick reviewed (commented):
Upon looking at this again, I wanted to move some of the board specific defines into platformio.ini rather than be in main.cpp, and with the extra led defines it made sense to do it now. Also, I know for certain the XAIO SAMD21 board LED is active low (and don't have my XIAO RP2040 board immediately at hand, but believe it is the same also), so the LEDs actually behave the exact opposite to how you wanted them, so added support for that, allowing for the PICO to remain unchanged. If you could double check this is still working for you, and respond to the other comments, that would be great.
